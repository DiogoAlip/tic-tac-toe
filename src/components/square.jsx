export const Square = ({ children, isSelected, updateBoard, index }) => {
	const className = `square ${isSelected ? "is-selected" : ""}`;
	return (
		// rome-ignore lint/a11y/useKeyWithClickEvents: <explanation>
		<div data-testid="Square-id" onClick={updateBoard} className={className}>
			{children}
		</div>
	);
};
