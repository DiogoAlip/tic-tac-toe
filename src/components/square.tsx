import React from "react";

type Props = {
  children?: JSX.Element | number;
  isSelected?: boolean;
  updateBoard?: () => void;
  index?: number;
};

export const Square = ({ children, isSelected, updateBoard, index }: Props) => {
  const className = `square ${isSelected ? "is-selected" : ""}`;
  return (
    <React.Fragment>
      {/** rome-ignore lint/a11y/useKeyWithClickEvents: <explanation>*/}
      <div data-testid="Square-id" onClick={updateBoard} className={className}>
        {children}
      </div>
    </React.Fragment>
  );
};
