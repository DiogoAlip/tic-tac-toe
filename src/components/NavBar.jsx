import { Square } from "./square";
import { marks } from "../constans";
import { useState } from "react";

export function NavBar({board, setBoard, turn, TURNS}){
    const [modal, setModal] = useState(false)
    const activateModal = () => {
        setModal(!modal)
    }

    const simbolChanger = (number) => {
        const newBoard = board.map((element)=>{
            if(element == turn){element = marks[number]}
            
        })
        setBoard(newBoard)

        for( let i in TURNS){
            if(i == turn){
                i = marks[number]
            }
            console.log(i)
        }
        activateModal()
    }

    return <div className="nav">
        <button onClick={activateModal}>Super Mishi</button>
        {modal && <section className="character-selector">
            {
                marks.map((_,index) => {
                    return (<Square key={index}>
                        <img 
                            className="table_element"
                            src={marks[index]}
                            alt=""
                            onClick={simbolChanger}/>
                    </Square>)
                })
            }
        </section>}
    </div>
}