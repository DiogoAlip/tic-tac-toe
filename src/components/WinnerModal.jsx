import { Square } from "./square"
export const WinnerModal = ({winner, resetGame}) => {
    if (winner == null) return null

    const TextWinnner =  winner == false ? 'Empate' : 'Gano' 

    return (
        <section className="winner">
            <div className="text">
                <h2>
                    {TextWinnner}
                </h2>
    
                <header className="win">
                    {winner && 
                    <Square><img className="table_element" src={winner} alt="" /></Square>}
                </header>
    
                <footer>
                    <button onClick={resetGame}>Empezar de nuevo</button>
                </footer>
            </div>
        </section>
    )
}