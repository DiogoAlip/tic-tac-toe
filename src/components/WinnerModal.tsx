import { Square } from "./square";
export const WinnerModal = ({
  winner,
  resetGame,
}: {
  winner: null | React.FC<React.SVGProps<SVGSVGElement>>;
  resetGame: () => void;
}) => {
  if (winner == null) return null;

  const TextWinnner = winner === null ? "Empate" : "Gano";

  return (
    <section className="winner">
      <div className="text">
        <h2>{TextWinnner}</h2>

        <header className="win">
          {winner && (
            <Square>
              <img className="table_element" src={winner.toString()} alt="" />
            </Square>
          )}
        </header>

        <footer>
          <button type="button" onClick={resetGame}>
            Empezar de nuevo
          </button>
        </footer>
      </div>
    </section>
  );
};
