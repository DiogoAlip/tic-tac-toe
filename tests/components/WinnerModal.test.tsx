import { WinnerModal } from "../../src/components/WinnerModal";
import { render, screen, fireEvent } from "@testing-library/react";

describe("WinnerModal.jsx", () => {
  test("shouldnt return a component", () => {
    const { container } = render(<WinnerModal winner={null} />);
    expect(container.getElementsByClassName("winner")[0]).toBeFalsy();
  });

  test("should return a component in won state", () => {
    const won = true;
    const { container } = render(<WinnerModal winner={won} />);
    expect(container.getElementsByClassName("winner")[0]).toBeTruthy();

    expect(screen.getByRole("heading", { level: 2 }).innerHTML).toBe("Gano");

    expect(screen.getByTestId("Square-id")).toBeTruthy();

    expect(container.getElementsByClassName("win")).toBeTruthy();

    expect(screen.getByText("Empezar de nuevo")).toBeTruthy();
  });

  test("should return a component in lost state", () => {
    const lost = false;
    const { container } = render(<WinnerModal winner={lost} />);

    expect(screen.getByRole("heading", { level: 2 }).innerHTML).toBe("Empate");

    expect(container.getElementsByClassName("table_element")[0]).toBeFalsy();

    expect(container.getElementsByClassName("win")).toBeTruthy();
  });

  test("WinnerModal should return an event", () => {
    let variable = false;
    const { container } = render(
      <WinnerModal
        resetGame={() => {
          variable = true;
        }}
        winner={!variable}
      />
    );
    expect(container).toMatchSnapshot();
    fireEvent.click(screen.getByText("Empezar de nuevo"));
    expect(variable).toBe(true);
  });
});
