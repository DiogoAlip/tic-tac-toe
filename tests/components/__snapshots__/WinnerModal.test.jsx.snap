// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WinnerModal.jsx WinnerModal should return an event 1`] = `
<div>
  <section
    class="winner"
  >
    <div
      class="text"
    >
      <h2>
        Gano
      </h2>
      <header
        class="win"
      >
        <div
          class="square "
          data-testid="Square-id"
        >
          <img
            alt=""
            class="table_element"
            src="true"
          />
        </div>
      </header>
      <footer>
        <button>
          Empezar de nuevo
        </button>
      </footer>
    </div>
  </section>
</div>
`;
